frontend/packages/dev-console/src/components/topology/shapes/PodSet.tsx

```diff
@@ -0,0 +1,78 @@
+import * as React from 'react';
+import { PodStatus, calculateRadius } from '@console/shared';
+import { PodData } from '../topology-types';
+
+interface PodSetProps {
+  size: number;
+  data: PodData;
+}
+function getPodData(data: PodData) {
+  // Scaling no. of pods
+  if (data.currentRCPhase === 'Complete' && data.currentAvailableReplicas !== data.totalReplicas) {
+    return { innerData: null, outerData: data.current };
+  }
+
+  // Deploy
+  switch (data.strategy) {
+    case 'Recreate':
+      if (
+        data.previous.length === 0 &&
+        data.currentReplicas === data.totalReplicas &&
+        data.currentAvailableReplicas === data.totalReplicas
+      ) {
+        return { innerData: null, outerData: data.current };
+      }
+      return { innerData: data.current, outerData: data.previous };
+    default:
+      // Rolling
+      if (data.previous.length > 0) {
+        return { innerData: data.current, outerData: data.previous };
+      }
+      return { innerData: null, outerData: data.current };
+  }
+}
+
+function calculateInnerPodStatusRadius(outerPodStatusInnerRadius, outerPodStatusWidth) {
```

Comment:
Use fat arrow syntax to declare the function. and move this function to utils file.